### PR TITLE
chore(acvm): updated ProofSystemCompiler methods and marked old prove and verify d…

### PR DESCRIPTION
**Pull request by vezenovm**
# Related issue(s)

No issue as very minor comment and deprecation update. Should have been done when I updated the `acvm` for preprocessing originally.

# Description

## Summary of changes

(Describe the changes in this PR. Point out breaking changes if any.)

## Dependency additions / changes

(If applicable.)

## Test additions / changes

(If applicable.)

# Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [ ] I have reviewed the changes on GitHub, line by line.
- [ ] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

File: acvm/src/lib.rs [hunk 1]
```diff
@@ -142,42 +142,39 @@ pub trait ProofSystemCompiler {
...
     ) -> bool;
 
+    /// Returns the number of gates in a circuit
     fn get_exact_circuit_size(&self, circuit: Circuit) -> u32;
 
+    /// Generates a proving and verification key given the circuit description
+    /// These keys can then be used to construct a proof and for its verification
     fn preprocess(&self, circuit: Circuit) -> (Vec<u8>, Vec<u8>);
 
+    /// Creates a Proof given the circuit description and the witness values.
```
Comment (kevaundray) [inline acvm/src/lib.rs:167]: nit: given th circuit description, the initial witness values and the proving key

File: acvm/src/lib.rs [hunk 2]
```diff
@@ -142,42 +142,39 @@ pub trait ProofSystemCompiler {
...
+    /// It is important to note that the intermediate witnesses for black box functions will not generated
+    /// This is the responsibility of the proof system.
     fn prove_with_pk(
         &self,
         circuit: Circuit,
         witness_values: BTreeMap<Witness, FieldElement>,
         proving_key: Vec<u8>,
     ) -> Vec<u8>;
 
+    /// Verifies a Proof, given the circuit description.
```
Comment (kevaundray) [inline acvm/src/lib.rs:177]: nit: same as above, we give it more than just the circuit description

**Review by kevaundray — COMMENTED** [review]
LGTM -- some small nits

**Comment by vezenovm** [comment]
> LGTM -- some small nits

Added some more info

**Review by kevaundray — APPROVED** [review]

**Pull request merged**